File: project/src/main/java/com/moyeoba/project/token/data/JwtPayload.java
package com.moyeoba.project.token.data;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Base64;
import java.util.Calendar;
import java.util.Date;

@Getter
@NoArgsConstructor
public class JwtPayload {
    private Integer validTime;     //분
    private String exp;     //expiration
    private String uid;     //user id

    public JwtPayload(Long id, Integer validTime) {
        this.validTime = validTime;
        Calendar calendar = Calendar.getInstance();
        calendar.add(Calendar.MINUTE, validTime);
        Date expiration = new Date(calendar.getTimeInMillis());

        uid = Long.toString(id);
        exp = Long.toString(expiration.getTime());
    }

    public String toJson() {
        ObjectMapper mapper = new ObjectMapper();
        String payloadJson = null;
        try {
            payloadJson = mapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }

        return payloadJson;
    }
    public String toJsonBase64() {
        String json = this.toJson();
        return Base64.getEncoder().encodeToString(json.getBytes());
    }

    public static JwtPayload getFromJson(String jsonStr) {
        JwtPayload payload = null;
        try {
            ObjectMapper mapper = new ObjectMapper();
            payload = mapper.readValue(jsonStr, JwtPayload.class);
        } catch (Exception e) {
            e.printStackTrace();
        }

        return payload;
    }
}

File: src/main/kotlin/com/moyeoba/moyeoba/jwt/token/JwtPayload.kt
package com.moyeoba.moyeoba.jwt.token

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import java.util.*


class JwtPayload() {
    lateinit var uid: String
        private set
    var validTime: Long = 0
        private set
    lateinit var exp: String
        private set


    constructor(uid: String, validTime: Long) : this() {
        this.uid = uid
        this.validTime = validTime

        val calendar = Calendar.getInstance()
        calendar.add(Calendar.SECOND, validTime.toInt())
        val expiration: Date = Date(calendar.timeInMillis)

        exp = expiration.time.toString()
    }


    fun toJson(): String? {
        val mapper = ObjectMapper()
        return mapper.writeValueAsString(this)
    }

    fun toJsonBase64(): String {
        val json = this.toJson()
        return Base64.getEncoder().encodeToString(json!!.toByteArray())
    }

    companion object {
        fun getFromJson(jsonStr: String): JwtPayload {
            val mapper = ObjectMapper()
            return mapper.readValue(jsonStr, JwtPayload::class.java)
        }
    }


}